frontend/src/components/modals/WarnThirdParty.tsx

```diff
@@ -0,0 +1,78 @@
+import { ConfirmModal } from '@decky/ui';
+import { FC, useEffect, useState } from 'react';
+import { FaExclamationTriangle } from 'react-icons/fa';
+
+import { WarnThirdPartyType } from '../../utils/globalTypes';
+import TranslationHelper, { TranslationClass } from '../../utils/TranslationHelper';
+
+interface WarnThirdPartyProps {
+  seconds?: number;
+  type: WarnThirdPartyType;
+  onOK(): void;
+  onCancel(): void;
+  closeModal?(): void;
+}
+
+const WarnThirdParty: FC<WarnThirdPartyProps> = ({ seconds = 5, type, onOK, onCancel, closeModal }) => {
+  const [waitTimer, setWaitTimer] = useState(seconds);
+
+  useEffect(() => {
+    // exit early when we reach 0
+    if (waitTimer <= 0) return;
+
+    // save intervalId to clear the interval when the
+    // component re-renders
+    const intervalId = setInterval(() => {
+      setWaitTimer(waitTimer - 1);
```

Comment:
Wouldn't set timeout work better for this then? Unless I'm misunderstanding?

Reply:
Yeah, fair enough. Not really that skilled with react.

Reply:
Reading this now, no, because I'm referencing the timeout on the translation on line 58 to show the remaining time ticking down to give additional feedback to the user, so I can't see how I would be doing this with one single timeout tick compared to one for each second. It was like 6 months from when I wrote this. Let me know if I'm missing something.